src/utils/internalLinks.js

```diff
@@ -61,9 +62,23 @@ export const getLinkType = (url: string, realm: string): LinkType => {
   return 'home';
 };
 
-/** PRIVATE -- exported only for tests. */
-export const transformToEncodedURI = (string: string): string =>
-  string.replace(/\.\d/g, (match, p1, p2, offset, str) => `%${match[1]}`);
+/** Decode MediaWiki-style dot-encoding. */
+// The Zulip server and webapp use this encoding for names with special
+// characters in narrow-links.
+export const decodeHashComponent = (string: string): string => {
+  try {
+    return decodeURIComponent(string.replace(/\./g, '%'));
```

Comment:
A pointer to `decodeHashComponent` in `static/js/hash_util.js` would be a helpful one-line comment.

(The same probably goes for some of the existing functions below, oops. But also the value is greater here, because the implementation shares so much verbatim.)